test-integration/operator-tests/src/test/java/org/infinispan/operator/security/CustomAuthTest.java

```diff
@@ -0,0 +1,63 @@
+package org.infinispan.operator.security;
+
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.Base64;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.http.entity.ContentType;
+import org.infinispan.Caches;
+import org.infinispan.Infinispan;
+import org.infinispan.Infinispans;
+import org.infinispan.TestServer;
+import org.infinispan.identities.Credentials;
+import org.infinispan.identities.Identities;
+import org.junit.jupiter.api.BeforeAll;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+
+import cz.xtf.builder.builders.RouteBuilder;
+import cz.xtf.builder.builders.SecretBuilder;
+import cz.xtf.client.Http;
+import cz.xtf.core.http.Https;
+import cz.xtf.junit5.annotations.CleanBeforeAll;
+import io.fabric8.kubernetes.api.model.Secret;
+import io.fabric8.openshift.api.model.Route;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@CleanBeforeAll
+public class CustomAuthTest extends AuthTest {
+   private static final Infinispan infinispan = Infinispans.customAuth();
+
+   @BeforeAll
+   public static void deploy() throws IOException {
+      testServer.deploy();
+
+      appName = infinispan.getClusterName();
+      host = openShift.generateHostname(appName);
+
+      Secret secret = new SecretBuilder("connect-secret").addData("identities.yaml", Paths.get("src/test/resources/secrets/identities.yaml")).build();
+      Route route = new RouteBuilder(appName).forService(appName).targetPort(11222).exposedAsHost(host).build();
+
+      openShift.secrets().create(secret);
+      openShift.customResource(infinispanContext).create(openShift.getNamespace(), infinispan.getInputStream());
+      openShift.createRoute(route);
+
+      openShift.waiters().areExactlyNPodsReady(2, "clusterName", appName).timeout(TimeUnit.MINUTES, 5).waitFor();
```

Comment:
This is waiting for 2 pods to be ready, right? Which 2 pods? 1 for the operator and 1 for infinispan server? Or 2 for infinispan server? I think it's the former but I'm unsure...

Reply:
It's waiting for two Infinispan pods. Test suite assumes that operator is already up and running in different namespace before test execution. Would be nicer to make this waiting function part of the `Infinispan` so it deduces the amount of pod that it should wait on from the CR but writing more tests is higher priority right now.

Reply:
Rather than waiting for 2 pods, could you wait for a cluster of 2 pods to be formed? That's what `wellFormed` status is about. It verifies that a cluster of N nodes (as defined in `.spec.replicas`) has been created. That's a better condition to wait for.

Reply:
That's a good idea. This way at least we verify that cluster forms in all tested cases of deployment.